test/val/val_instructions_test.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2017 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Validation tests for illegal instructions
+
+#include <sstream>
+#include <string>
+#include <tuple>
+
+#include "gmock/gmock.h"
+#include "val_fixtures.h"
+
+using ::testing::HasSubstr;
+
+using ValidateIns = spvtest::ValidateBase<std::string>;
+
+namespace {
+
```

Comment:
High level thing to fix: There's a lot of repetition here between the three cases.

It's preferable to make a function to return the std::string containing the common preamble assembly text: everything up to the actual instruction being tested.  That will shorten the test, and keep each focused on what's actually different about it.

(Google style guide forbids non-POD variables with static duration.  So just use a function to return the std::string).

Alternately, you can use  gtest parameterized test infrastructure to present a bunch of cases as a collection of structs.  E.g. see the TEST_P and INSTANTIATE_TEST_CASE_P at https://github.com/KhronosGroup/SPIRV-Tools/blob/master/test/text_to_binary.constant_test.cpp#L33
In that case you'd have to make sure to capture the differences in the operands for the various instructions.  So a test case is both the instruction name, and its different operand lists.